### ISSUE TITLE
TreeRelatedFieldListFilter does not work

### ISSUE THREAD

**Issue by pawelmarkowski** [django-mptt/django-mptt#669]
Link generated in right side filter panel is incorrect. mptt '0.9.1', django '2.1'

list_filter = (("catalogue", TreeRelatedFieldListFilter),)

_product/?catalogue__id__exact=22_ instead of _product/?catalogue__id__inhierarchy=22_ which is correct. If we replace exect to inhierarchy it works fine.

**Comment by pawelmarkowski** [comment]
#606 it looks like a root cause.

**Comment by ezzra** [comment]
> #606 it looks like a root cause.

yes, #606 is fixed for me too, when I change the link manually, and the link also does the correct query

**Comment by convers39** [comment]
Hi there! Is this issue fixed? It seems didn't work on my app too. It supposes to return related instances including all descendants of that related instance right? 
For example, item -fk-> category, and in item list_filter add TreeRelatedFieldListFilter for category, then when you search items of category A, the descendants from category B, C... are also going to show up.

**Comment by matthiask** [comment]
It may not be; I'd appreciate a pull request.

**Comment by matthiask** [comment]
This was fixed in the same commit that fixed #579. The filter correctly generates `__inhierarchy` links and the `selected` state should be correct. If you can reproduce this on a current version please open a new issue with details. See also #833.
